client/v2/limits.go

```diff
@@ -0,0 +1,75 @@
+package v2
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+const (
+	// HeaderRateLimit is the (draft07) recommended header from the IETF
+	// on rate limiting.
+	HeaderRateLimit = "RateLimit"
+	// The value of the header is formatted "limit=X, remaining=Y, reset=Z".
+	// Where:
+	//   - X is the maximum number of requests allowed in the window
+	//   - Y is the number of requests remaining in the window
+	//   - Z is the number of seconds until the limit resets
+	HeaderRateLimitFmt = "limit=%d, remaining=%d, reset=%d"
+
+	// HeaderRetryAfter is the RFC7231 header used to indicate when a client should
+	// retry requests in UTC time.
+	HeaderRetryAfter = "Retry-After"
+)
+
+var rng *rand.Rand
+
+func init() {
+	// initialize the random number generator
+	rng = rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Is there a reason not to use the rand package-level functions?

Reply:
I honestly forgot `rand.Float64()` was there! Thanks!